jsonnet/kube-thanos/kube-thanos-store-shards.libsonnet

```diff
@@ -16,35 +16,76 @@ function(params)
   assert std.isNumber(config.shards) && config.shards >= 0 : 'thanos store shards has to be number >= 0';
 
   { config:: config } + {
-    ['shard' + i]: store(config {
-      name+: '-%d' % i,
-      commonLabels+:: { 'store.observatorium.io/shard': 'shard-' + i },
-    }) {
-      statefulSet+: {
-        spec+: {
-          template+: {
-            spec+: {
-              containers: [
-                if c.name == 'thanos-store' then c {
-                  args+: [
-                    |||
-                      --selector.relabel-config=
-                        - action: hashmod
-                          source_labels: ["__block_id"]
-                          target_label: shard
-                          modulus: %d
-                        - action: keep
-                          source_labels: ["shard"]
-                          regex: %d
-                    ||| % [config.shards, i],
-                  ],
-                } else c
-                for c in super.containers
-              ],
+    shards: {
```

Comment:
Adding another level of nesting here breaks the contract with consumers, but so does removing the per-shard ServiceMonitors, which is kind of the whole point of this PR 🤔 

I'm not sure what the maintainers will think of this yet, but I figured it was worth starting the conversation.

Reply:
Can we solve this without introducing another level of nesting? This is more breaking than merging service monitors into one. And I like the idea we have to do the same for receiver monitors I believe.

Reply:
We can indeed. I just realised that I didn't explain why I did this: I thought that it would look weird to have shard0, shard1, etc alongside some non-shard object without another level of nesting.

@kakkoyun with that in mind, do you still want to remove the nesting? The return value would look like:

```json
{
  "shard0": {
    "statefulSet": {}
    ...
  },
  ...
  "serviceMonitor": {},
}
```

Reply:
Aha! Now it makes sense. Sorry I didn't realize that. Yes, let's keep it 👍